packages/ws-server/src/webSocketServer.ts

```diff
@@ -63,10 +68,53 @@ function getMaxConnectionTTL() {
     return parseInt(process.env.WS_MAX_CONNECTION_TTL || '300000');
 }
 
-app.ws.use((ctx) => {
+// function to get maximum allowed connections from a single IP
+function getLimitPerIp() {
+    return parseInt(process.env.CONNECTION_LIMIT_PER_IP || '10');
```

Comment:
This should be from a constant so you're not looking up and converting each time.
If you're not exposing publicly not sure that a getter is needed